devito/passes/iet/orchestration.py

```diff
@@ -177,45 +60,43 @@ def _make_waitlock(self, iet, sync_ops, *args):
     def _make_withlock(self, iet, sync_ops, pieces, root):
         locks = sorted({s.lock for s in sync_ops}, key=lambda i: i.name)
 
-        threads = self.__make_threads(value=min(i.size for i in locks))
+        npthreads = min(i.size for i in locks)
 
         preactions = []
         postactions = []
         for s in sync_ops:
             imask = [s.handle.indices[d] if d.root in s.lock.locked_dimensions else FULL
                      for d in s.target.dimensions]
 
-            preactions.append(List(body=[
-                BlankLine,
-                List(header=self._P._map_update_wait_host(s.target, imask,
-                                                          SharedData._field_id)),
-                DummyExpr(s.handle, 1)
-            ]))
+            update = self._P._map_update_wait_host(s.target, imask, SharedData._field_id)
+            update = List(header=update)
+
+            preactions.append(List(body=[BlankLine, update, DummyExpr(s.handle, 1)]))
             postactions.append(DummyExpr(s.handle, 2))
         preactions.append(BlankLine)
         postactions.insert(0, BlankLine)
 
-        # Turn `iet` into an ElementalFunction so that it can be
-        # executed asynchronously by `threadhost`
+        # Turn `iet` into a ThreadFunction so that it can be executed
+        # asynchronously by a child host thread
```

Comment:
What's the difference between a 'child host thread' and a pthread?

Remark: Given the number of different uses for thread (pthreads, openmp, openacc, etc) perhaps use the associated  qualifier (ie pthread, OpenMP threads, etc)/

Reply:
> What's the difference between a 'child host thread' and a pthread?

no difference. I will clarify